### PR TITLE
[MO] update ROIAlign op to opset9

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - [x] add `aligned_mode` attribute
 - [x] update ROIAligned MO implementation
 - [x] add MO unit tests for ROIAlign-9

### Tickets:
 - 83653

### CODE REVIEW

File: tools/mo/unit_tests/mo/ops/roialign_test.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import unittest
+from openvino.tools.mo.front.common.partial_infer.utils import int64_array, float32_array
+
+import numpy as np
+from openvino.tools.mo.graph.graph import Node
+from openvino.tools.mo.ops.roialign import ROIAlign
+from unit_tests.utils.graph import build_graph
+
+
+class TestROIAlignOps(unittest.TestCase):
```
Comment (mitruska) [inline tools/mo/unit_tests/mo/ops/roialign_test.py:13]: There are additional tests for MO ONNX ROIAlign:
https://github.com/openvinotoolkit/openvino/blob/252c72146b8c7ec054bc27f001bbdef52c783f51/tests/layer_tests/onnx_tests/test_roi_align.py#L107-L109
Could you enable it (`@pytest.mark.precommit`) at least for one CI cycle?
Also you can update these tests if needed, and please ensure they won't go by default MO path, so with ONNX FE (onnx importer).

**Review by mitruska — CHANGES_REQUESTED** [review]

**Review by tomdol — APPROVED** [review]

File: tools/mo/openvino/tools/mo/front/onnx/roialign_ext.py [hunk 2]
```diff
@@ -17,8 +17,14 @@ def extract(cls, node):
         output_width = onnx_attr(node, 'output_width', 'i', default=1)
         sampling_ratio = onnx_attr(node, 'sampling_ratio', 'i', default=0)
         spatial_scale = onnx_attr(node, 'spatial_scale', 'f', default=1.0)
-
-        ROIAlign.update_node_stat(node, {'pooled_h': output_height, 'pooled_w': output_width,
-                                         'sampling_ratio': sampling_ratio, 'spatial_scale': spatial_scale,
-                                         'mode': mode})
+        onnx_opset_version = get_onnx_opset_version(node)
+        if onnx_opset_version >= 16:
+            aligned_mode = onnx_attr(node, 'coordinate_transformation_mode', 's', default=b'half_pixel').decode()
```
Comment (mitruska) [inline tools/mo/openvino/tools/mo/front/onnx/roialign_ext.py:22]: It won't work for onnx ROIAlign-16 `coordinate_transformation_mode` set to "output_half_pixel".
We need to map it to one of the openvino ROIAlign-9 `aligned_mode` options.

Please take a look at support for ONNX ROIAlign-16 from new ONNX FE (onnx importer):
https://github.com/openvinotoolkit/openvino/blob/5fa669785c555838df8b3cdc3fe24a644c91d59c/src/frontends/onnx/frontend/src/op/roi_align.cpp#L64-L66

File: tests/layer_tests/onnx_tests/test_roi_align.py [hunk 3]
```diff
@@ -47,8 +63,11 @@ def create_net(self, input_shape, rois_shape, indices_shape, output_shape,
             [output],
         )
 
+        operatorsetid = OperatorSetIdProto()
+        operatorsetid.domain = ""
+        operatorsetid.version = 10
```
Comment (mitruska) [inline tests/layer_tests/onnx_tests/test_roi_align.py:68]: What about tests for ONNX ROIAlign-16?

File: tests/layer_tests/common/layer_test_class.py [hunk 4]
```diff
@@ -60,6 +60,8 @@ def _test(self, framework_model, ref_net, ie_device, precision, ir_version, temp
 
         if use_new_frontend:
             mo_params["use_new_frontend"] = True
+        else:
+            mo_params["use_legacy_frontend"] = True
```
Comment (mitruska) [inline tests/layer_tests/common/layer_test_class.py:64]: I suggested this line to enable triggering tests for legacy frontend, but it also has impact on default frontend for layer tests. I think we should set `use_new_frontend` default `True`, to keep testing new frontend as default. 
https://github.com/openvinotoolkit/openvino/blob/9a2f2e446189d396fd27dbc6b7d67520ae00b47b/tests/layer_tests/common/layer_test_class.py#L33

**Review by mitruska — CHANGES_REQUESTED** [review]

File: tests/layer_tests/onnx_tests/test_roi_align.py (hunk 3 above)
Comment (bszmelcz) [reply tests/layer_tests/onnx_tests/test_roi_align.py:68]: I will add them

File: tools/mo/openvino/tools/mo/ops/roialign.py [hunk 5]
```diff
@@ -53,7 +65,9 @@ def infer(node):
         assert node.has_valid('mode'), '"mode" attribute is not set for node "{}"'.format(node_name)
         assert node.mode in ['avg', 'max'], \
             '"mode" attribute range of values is ["avg", "max"], got {} for node "{}"'.format(node.mode, node_name)
-
+        if node.get_opset() == 'opset9':
+            assert node.aligned_mode in ['asymmetric', 'half_pixel_for_nn', 'half_pixel'], \
+                '"aligned_mode" attribute range of values is ["asymmetric", "half_pixel_for_nn", "half_pixel"]'
```
[the remark this answers is not in the document]
Comment (bszmelcz) [reply tools/mo/openvino/tools/mo/ops/roialign.py:70]: Yeah, got it.

**Review by mitruska — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**